pytreex/block/read/conllu.py

```diff
@@ -46,50 +46,63 @@ def process_document(self, filename):
         last_node = root
         nodes = [root]
         parents = [0]
+        comment = ''
         
         for line in fh:
             
             # Strip newline character (\n or \r\n)
             line = line.rstrip('\r\n')
-            
-            # Skip empty lines and comments before start of sentence
-            if len(nodes)==1 and (not line or line.startswith('#')): continue
-            
+
             # Empty line as a end of sentence
             if not line:
+                # Ignore (multiple) empty lines before start of sentence (invalid CoNLL-U)
+                if len(nodes)==1:
+                    continue
+
+                # Rehang to correct parents and save nonempty comment to root
                 for i in xrange(1,len(nodes)):
                     nodes[i].parent = nodes[parents[i]]
+                if len(comment):
+                    root.set_attr('wild/comment', comment)
+
+                # Prepare a new bundle
                 bundle = doc.create_bundle()
                 zone = bundle.create_zone(self.language, self.selector)
                 root = zone.create_atree()
                 last_node = root
                 nodes = [root]
                 parents = [0]
-                continue
+                comment = ''
             
-            columns = line.split('\t')
+            # Comment
+            elif line[0] == '#':
+                comment = comment + line[1:] + "\n"
+
+            # A normal line with one token
+            else:
+                columns = line.split('\t')
             
-            # TODO: multi-word tokens
-            if '-' in columns[0]: continue
+                # TODO: multi-word tokens
+                if '-' in columns[0]: continue
```

Comment:
Thanks, I'll fix it. I still don't have pylint installed. (origin: @foxik https://github.com/ufal/rh_nntagging/blob/master/lib/conllu.py#L66)